src/sage/crypto/key_exchange/diffie_hellman.py

```diff
@@ -0,0 +1,314 @@
+        - ``sk`` -- secret key
+
+        EXAMPLES::
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH = DiffieHellman(17, 3)
+            sage: DH.compute_shared_secret(13, 11)
+            4
+        """
+        return self._field(pk**sk)
+
+    def subgroup_size(self) -> Integer:
+        """
+        Calculates the size of the subgroup of `\\GF{p}` generated by
+        ``self.generator()``.
+
+        EXAMPLES:
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH = DiffieHellman(47, 2)
+            sage: DH.subgroup_size()
+            23
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH = DiffieHellman(47, 5)
+            sage: DH.subgroup_size()
+            46
+        """
+        return self._g.multiplicative_order()
+
+    def __len__(self):
+        """
+        Calculates the size of the subgroup of `\\GF{p}` generated by
+        ``self.generator()``. This is a wrapper around `subgroup_size`.
+        """
+        return int(self.subgroup_size())
+
+    def __eq__(self, other):
+        """
+        Check if two Diffie-Hellman instances have the same parameter set.
+
+        TESTS::
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH1 = DiffieHellman(5, 2)
+            sage: DH2 = DiffieHellman(5, 2)
+            sage: DH1 == DH2
+            True
+        """
+        return self.parameters() == other.parameters()
+
+    def __hash__(self):
+        """
+        Compute the hash value of a Diffie-Hellman instance.
+
+        TESTS::
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH1 = DiffieHellman(7, 3)
+            sage: DH2 = DiffieHellman(7, 3)
+            sage: s = set([DH1, DH2])
+            sage: len(s)
+            1
+        """
+        return hash((self._p, self._g))
+
+    def _repr_(self):
+        """
+        Get the string representation of the Diffie-Hellman instance.
+
+        TESTS::
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH = DiffieHellman(7, 3)
+            sage: DH
+            Diffie-Hellman key exchange over Finite Field of size 7 with generator 3
+        """
+        return f'Diffie-Hellman key exchange over {self._field} with generator {self._g}'
+
+    def _latex_(self):
+        r"""
+        Get the LaTeX representation of the Diffie-Hellman instance.
+
+        TESTS::
+
+            sage: from sage.crypto.key_exchange.diffie_hellman import DiffieHellman
+            sage: DH = DiffieHellman(7, 3)
+            sage: latex(DH)
+            \text{Diffie-Hellman key exchange over }\Bold{F}_{7}\text{ with generator }3
+        """
+        return f'\\text{{Diffie-Hellman key exchange over }}{self._field._latex_()}\\text{{ with generator }}{self._g}'
```

Comment:
Same here for 80 cols format
```suggestion
        return f'\\text{{Diffie-Hellman key exchange over }}' \
               f'{self._field._latex_()}\\text{{ with generator }}{self._g}'
```